pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -260,14 +264,23 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
                 "Segment metadata file (as multipart/form-data) is required for METADATA upload mode",
                 Response.Status.BAD_REQUEST);
           }
-          if (StringUtils.isEmpty(downloadURI)) {
-            throw new ControllerApplicationException(LOGGER, "Download URI is required for METADATA upload mode",
+          if (StringUtils.isEmpty(sourceDownloadURIStr)) {
+            throw new ControllerApplicationException(LOGGER,
+                "Source download URI is required in header field 'DOWNLOAD_URI' for METADATA upload mode",
                 Response.Status.BAD_REQUEST);
           }
-          moveSegmentToFinalLocation = false;
+          // override copySegmentToFinalLocation if override provided in headers:COPY_SEGMENT_TO_DEEP_STORE
+          // else set to false for backward compatibility
+          String copySegmentToDeepStore =
+              extractHttpHeader(headers, FileUploadDownloadClient.CustomHeaders.COPY_SEGMENT_TO_DEEP_STORE);
+          if (copySegmentToDeepStore != null) {
+            copySegmentToFinalLocation = Boolean.parseBoolean(copySegmentToDeepStore);
+          } else {
+            copySegmentToFinalLocation = false;
+          }
```

Comment:
```suggestion
          copySegmentToFinalLocation = Boolean.parseBoolean(copySegmentToDeepStore);
```
should do the trick. null is mapped to false.